libmamba/include/mamba/solver/solution.hpp

```diff
@@ -112,35 +113,27 @@ namespace mamba::solver
                 action
             );
         }
-    }
 
-    // TODO(C++20): Poor man's replacement to range filter transform
-    template <typename Iter, typename UnaryFunc>
-    void for_each_to_remove(Iter first, Iter last, UnaryFunc&& func)
-    {
-        for (; first != last; ++first)
+        template <typename Range>
```

Comment:
As we are now in C++20, I would prefer if any new/changed template type parameter be constrained by C++ concepts, in particular in situations where we already have the concept in the STL. Here it would be one of the [range concepts](https://en.cppreference.com/w/cpp/ranges.html) (see "range concepts" section. Not sure which one is expected here.

Same remark to other functions below.